gen_docker_compose.py

```diff
@@ -72,16 +72,19 @@ def generate_compose_file(
             sys.exit(1)
 
         template_args = {}
-        assert 'versions' in self.config
+
+        if 'versions' not in self.config:
+            raise RuntimeError("Bad config, no [versions] section found")
+
         for k, v in self.config['versions'].items():
             template_args[k] = v
 
         for k, v in self.config[f'device_{device_type}'].items():
             template_args[k] = v
 
         output = template.render(**template_args)
-        open(output_file, 'w').write(output)
-
+        with open(output_file, 'w') as template_output:
```

Comment:
Actually both removed and added code does the same thing. File is closed automatically in previous code too. The current approach is better though so in future if we need further processing on the file then we have a variable to access the file.

Reply:
ahh ok! good to know for future. It turned out the issue was actually with the `data` in the directory at the top - but didn't change this back